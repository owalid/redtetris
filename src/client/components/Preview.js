import React from 'react';
import Card from '@material-ui/core/Card';
import PreviewPiece from './PreviewPiece';
import OtherPlayerGrid from './OtherPlayerGrid';

const boxProps = {
  bgcolor: 'background.paper',
  bordercolor: 'text.primary',
  border: 1,
  borderradius: "borderRadius",
  style: { backgroundColor: 'white', width: '100vw'},
};

const Preview = ({ mapsGamePreview, isAlone, isVisitor, score, sheet }) => 
  <>
    <div className="d-flex jcnt--start fdir--column">
      {!isVisitor && sheet && (
          <div className="aself--fstart width-100 test--container-not-visitor">
            <Card className="pt-3" {...boxProps} style= {{ width: '30vw' }} variant="outlined">
              <p className="pl-2">Prochaine piece:</p>
              <PreviewPiece sheet={sheet}/>
            </Card>
          </div>
      )}
      <div className="aself--fstart">
        <Card {...boxProps} style= {{ width: '30vw', height: '5vh' }} variant="outlined">
          <p className="pl-3">
            Score: <span className="bold">{ score }</span>
          </p>
        </Card>
      </div>
    </div>
    <div className="container-grid">
      {
        mapsGamePreview.map(game => (
          <div key={game} className="p-2">
            <OtherPlayerGrid
              isAlone={isAlone}
              mapGamePreview={game}
              />
          </div>
        ))
      }
    </div>
  </>
    
export default Preview
