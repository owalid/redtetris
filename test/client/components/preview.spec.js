import React from "react";
import _ from 'lodash'
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react'
import { shallow } from 'enzyme';
import Preview from '../../../src/client/components/Preview'
import { player_instance1, room1 } from "../helpers/data";

describe('Preview component', () => {

  const Wrapper = () => {
    <Preview />
  }

  it('Is exists', () => {
    const wrapper = shallow(<Wrapper />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Can mount', () => {
    const Wr = () => (
      <Preview
        mapsGamePreview={_.filter(room1.players, item => item.uuid !== player_instance1.uuid && !item.visitor)}
        isVisitor={false}
        isAlone={false}
        score={room1.players[player_instance1.uuid].score}
        sheet={room1.players[player_instance1.uuid].sheets[0]}
        uuidRoom={room1.channel}
      />
    )
    const w = render(<Wr />);
    expect(w).not.toBeNull()
  })

  it('Element not visitor', () => {
    const Wr = () => (
      <Preview
        mapsGamePreview={_.filter(room1.players, item => item.uuid !== player_instance1.uuid && !item.visitor)}
        isVisitor={true}
        isAlone={false}
        score={room1.players[player_instance1.uuid].score}
        sheet={room1.players[player_instance1.uuid].sheets[0]}
        uuidRoom={room1.channel}
      />
    )

    const { container } = render(<Wr />);
    const container_not_visitor = container.querySelector('.test--container-not-visitor')

    expect(container_not_visitor).toBeNull()
  })
});