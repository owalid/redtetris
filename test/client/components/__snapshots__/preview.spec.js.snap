// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview component Is exists 1`] = `""`;
